cmd/evm/main.go

```diff
@@ -125,7 +125,7 @@ func run(ctx *cli.Context) error {
 	glog.SetV(ctx.GlobalInt(VerbosityFlag.Name))
 
 	db, _ := ethdb.NewMemDatabase()
-	statedb, _ := state.New(common.Hash{}, db)
+	statedb, _ := state.New(common.Hash{}, state.NewDatabase(db))
```

Comment:
This naming convention of `statedb` in this common pattern I find personally misleading. `state.New` returns a `*StateDB` struct, and that's why, but I think even that struct name is misleading because it's really more than just a db... I'd say it's more a `StateTrieManager`. Just a heads up. 

Maybe wew can come up with a better naming convention.